daemon/SecretsImpl/secretsrequestprocessor.cpp

```diff
@@ -2042,10 +2042,6 @@ Sailfish::Secrets::Daemon::ApiImpl::RequestProcessor::getCollectionSecret(
     } else if (!m_authenticationPlugins.contains(collectionAuthenticationPluginName)) {
         return Sailfish::Secrets::Result(Sailfish::Secrets::Result::InvalidExtensionPluginError,
                                         QString::fromLatin1("No such authentication plugin available: %1").arg(collectionAuthenticationPluginName));
-    } else if (m_authenticationPlugins[collectionAuthenticationPluginName]->authenticationType() == Sailfish::Secrets::AuthenticationPlugin::ApplicationSpecificAuthentication
```

Comment:
This check was premature.  There can be cases where the collection's authentication plugin requires in-app authentication method, but user interaction is not necessary (e.g. if the collection is owned by the application and unlocked).  So I removed this premature check, and only perform this check in the cases where user interaction is deemed required (below).